scripts/WeaponFX.js

```diff
@@ -130,8 +130,28 @@ Hooks.on("createChatMessage", (data) => {
 
     const {weaponIdentifier, sourceToken} = messageMeta;
 
-    const macroName = weaponEffects[weaponIdentifier];
-    if (!macroName) return;
+    var macroName;
+
+    // weaponIdentifier can variously be either an object with Foundry item data, or a string
```

Comment:
Since `weaponIdentifier` can now _only_ be an object (from the change to messageParser.js line 85), I don't think the `instanceof` conditional is necessary; we can safely assume it's always a weapon object. So let's make the following changes:

* Rename `MessageInfo.weaponIdentifier` to `MessageInfo.weaponObject`
* Remove the `instanceof` conditional block entirely (including the `else` clause)
* JUST keep lines 139-147 as the whole check. That means we check the FX list by weapon name, and if that returns nothing, we check using LID.

Reply:
Hmm, I just re-read your note saying that `weaponIdentifier` might not be an object, I did miss that. I suppose that's fair here then. I'll ponder it a bit more.